# **Data Cleaning (chipotle.tsv):**

Columns:
1.   Order ID
2.   Quantity
3.  Item Name
4. Choice Description
5. Item Price





In [1]:
#import the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import re
import ast

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
tsv_file = '/content/drive/MyDrive/Colab Notebooks/Copy of Copy of chipotle.tsv'
data = pd.read_csv(tsv_file, sep='\t')

In [4]:
data

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [5]:
#creating a dataframe for the data
df=pd.DataFrame(data)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


In [7]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]


# 1. Missing Values:

Question: Check for missing values in each column (Order ID, Quantity, Item Name, Choice Description, Item Price). How should missing values be handled?


In [8]:
missing_values = df.isnull().sum()
print("Missing Values in Each Column:")
print(missing_values)

Missing Values in Each Column:
order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64


In [9]:
# Fill missing values in 'choice_description' with "No description"
df['choice_description'].fillna("None", inplace=True)
df.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                               None     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                               None     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                               None     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

In [10]:
#verifying If there is anyy null values after filling the Nan
missing_values = df.isnull().sum()
missing_values

order_id              0
quantity              0
item_name             0
choice_description    0
item_price            0
dtype: int64


# 2. Data Types:

Question: Verify the data types of each column. Do they align with their expected types, and should any adjustments be made?


In [11]:
#Converting the item_price to float data type
# Remove the dollar sign ($) and convert 'item_price' to float
# Assuming you want to replace '$' and handle NaN values
df['item_price'] = df['item_price'].replace('[\$,]', '', regex=True).replace('', np.nan).astype(float)

In [12]:
df.dtypes

order_id                int64
quantity                int64
item_name              object
choice_description     object
item_price            float64
dtype: object


# 3. Duplicated Entries:

Question: Identify and handle duplicated entries in the dataset. How might duplicates impact analysis, and what is the appropriate action?


In [13]:
#No of Rows before removing the duplicates
df.shape

(4622, 5)

In [14]:
# Finding duplicates based on specific columns (e.g., Order_Id and Item_Name ,Choice_Description)
df[df.duplicated(subset=['order_id', 'item_name' ,'choice_description'], keep=False)]

order_id  quantity         item_name  \
234        103         1     Steak Burrito   
238        103         1     Steak Burrito   
246        108         1       Canned Soda   
248        108         1       Canned Soda   
296        129         1     Steak Burrito   
...        ...       ...               ...   
4544      1806         1     Bottled Water   
4568      1817         1     Bottled Water   
4570      1817         1     Bottled Water   
4602      1827         1  Barbacoa Burrito   
4603      1827         1  Barbacoa Burrito   

                                     choice_description  item_price  
234   [Tomatillo Red Chili Salsa, [Rice, Black Beans...       11.75  
238   [Tomatillo Red Chili Salsa, [Rice, Black Beans...       11.75  
246                                      [Mountain Dew]        1.09  
248                                      [Mountain Dew]        1.09  
296   [Tomatillo Green Chili Salsa, [Rice, Cheese, G...       11.75  
...                                                 ...         ...  
4544                                               None        1.50  
4568                                               None        1.50  
4570                                               None        1.50  
4602                      [Tomatillo Green Chili Salsa]        9.25  
4603                      [Tomatillo Green Chili Salsa]        9.25  

[117 rows x 5 columns]

In [15]:
#Dropping the duplicates
df.drop_duplicates(subset=['order_id', 'item_name' , 'choice_description'], keep=False, inplace=True)
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                                  None        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                  None        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
...                                                 ...         ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75  

[4505 rows x 5 columns]

In [16]:
#No of rows after removing the duplicates
df.shape
#4505 rows(after removing the duplicates)

(4505, 5)


# 4. Quantity and Item Price:

Question: Examine the Quantity and Item Price columns. Are there any inconsistencies or anomalies that need correction?


In [17]:
df.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description  item_price  
0                                               None        2.39  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                               None        2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98  
6                                               None        1.69  
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.75  
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...        9.25  
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...        9.25

In [18]:
# Check for negative or zero values in the Quantity column
negative_quantity = df[df['quantity'] <= 0]
print("Rows with negative or zero Quantity values:")
print(negative_quantity)

Rows with negative or zero Quantity values:
Empty DataFrame
Columns: [order_id, quantity, item_name, choice_description, item_price]
Index: []


In [19]:
# Summary statistics for Quantity column
(df['quantity'].describe())

count    4505.000000
mean        1.076138
std         0.405580
min         1.000000
25%         1.000000
50%         1.000000
75%         1.000000
max        15.000000
Name: quantity, dtype: float64

In [20]:
# Summary statistics for Item Price column
(df['item_price'].describe())

count    4505.000000
mean        7.514337
std         4.243410
min         1.090000
25%         3.990000
50%         8.750000
75%         9.390000
max        44.250000
Name: item_price, dtype: float64

Item Price: Prices vary, ranging from 1.09 to 44.25, with a mean of 7.51


## 5. Choice Description:

Question: Analyze the Choice Description column. How should choices be handled, especially when there are multiple descriptions for a single item?


In [21]:
df['choice_description'].head(20)

0                                                  None
1                                          [Clementine]
2                                               [Apple]
3                                                  None
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...
5     [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...
6                                                  None
7     [Tomatillo Red Chili Salsa, [Fajita Vegetables...
8     [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...
9     [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...
10                                                 None
11    [Roasted Chili Corn Salsa, [Fajita Vegetables,...
12    [Roasted Chili Corn Salsa, [Rice, Black Beans,...
13    [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...
14                                                 None
15                                                 None
16    [Tomatillo-Green Chili Salsa (Medium), [Pinto ...
17    [Fresh Tomato Salsa (Mild), [Black Beans, 

In [22]:
#Splitting the Choice_Description column
split_columns = df['choice_description'].str.split(",", expand=True)


split_columns.columns = ['choice_description_1', 'choice_description_2', 'choice_description_3',
                         'choice_description_4', 'choice_description_5', 'choice_description_6',
                         'choice_description_7', 'choice_description_8', 'choice_description_9',
                         'choice_description_10']

df = pd.concat([df, split_columns], axis=1)

In [23]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  \
0                                                  None        2.39   
1                                          [Clementine]        3.39   
2                                               [Apple]        3.39   
3                                                  None        2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98   
...                                                 ...         ...   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75   

                  choice_description_1 choice_description_2  \
0                                 None                 None   
1                         [Clementine]                 None   
2                              [Apple]                 None   
3                                 None                 None   
4     [Tomatillo-Red Chili Salsa (Hot)         [Black Beans   
...                                ...                  ...   
4617               [Fresh Tomato Salsa                [Rice   
4618               [Fresh Tomato Salsa                [Rice   
4619               [Fresh Tomato Salsa   [Fajita Vegetables   
4620               [Fresh Tomato Salsa   [Fajita Vegetables   
4621               [Fresh Tomato Salsa   [Fajita Vegetables   

     choice_description_3 choice_description_4 choice_description_5  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    Rice               Cheese         Sour Cream]]   
...                   ...                  ...                  ...   
4617          Black Beans           Sour Cream               Cheese   
4618           Sour Cream               Cheese              Lettuce   
4619          Pinto Beans            Guacamole            Lettuce]]   
4620            Lettuce]]                 None                 None   
4621          Pinto Beans            Lettuce]]                 None   

     choice_description_6 choice_description_7 choice_description_8  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    None                 None                 None   
...                   ...                  ...                  ...   
4617              Lettuce          Guacamole]]                 None   
4618          Guacamole]]                 None                 None   
4619                 None                 None                 None   
4620                 None   

In [24]:
#dropping the Choice_Desciption column
df.drop('choice_description', axis=1, inplace=True)

In [25]:
df

order_id  quantity                              item_name  item_price  \
0            1         1           Chips and Fresh Tomato Salsa        2.39   
1            1         1                                   Izze        3.39   
2            1         1                       Nantucket Nectar        3.39   
3            1         1  Chips and Tomatillo-Green Chili Salsa        2.39   
4            2         2                           Chicken Bowl       16.98   
...        ...       ...                                    ...         ...   
4617      1833         1                          Steak Burrito       11.75   
4618      1833         1                          Steak Burrito       11.75   
4619      1834         1                     Chicken Salad Bowl       11.25   
4620      1834         1                     Chicken Salad Bowl        8.75   
4621      1834         1                     Chicken Salad Bowl        8.75   

                  choice_description_1 choice_description_2  \
0                                 None                 None   
1                         [Clementine]                 None   
2                              [Apple]                 None   
3                                 None                 None   
4     [Tomatillo-Red Chili Salsa (Hot)         [Black Beans   
...                                ...                  ...   
4617               [Fresh Tomato Salsa                [Rice   
4618               [Fresh Tomato Salsa                [Rice   
4619               [Fresh Tomato Salsa   [Fajita Vegetables   
4620               [Fresh Tomato Salsa   [Fajita Vegetables   
4621               [Fresh Tomato Salsa   [Fajita Vegetables   

     choice_description_3 choice_description_4 choice_description_5  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    Rice               Cheese         Sour Cream]]   
...                   ...                  ...                  ...   
4617          Black Beans           Sour Cream               Cheese   
4618           Sour Cream               Cheese              Lettuce   
4619          Pinto Beans            Guacamole            Lettuce]]   
4620            Lettuce]]                 None                 None   
4621          Pinto Beans            Lettuce]]                 None   

     choice_description_6 choice_description_7 choice_description_8  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    None                 None                 None   
...                   ...                  ...                  ...   
4617              Lettuce          Guacamole]]                 None   
4618          Guacamole]]                 None                 None   
4619                 None                 None                 None   
4620                 None                 None                 None   
4621                 None                 None                 None   

     choice_description_9 choice_description_10  
0                    None                  None  
1                    None                  None  
2                    None                  None  
3                    None                  None  
4                    None                  None  
...                   ...                   ...  
4617                 None                  None  
4618                 None                  None  
4619                 None                  None  
4620                 None                  None  
4621                 None                  N


## 6. Handling Special Characters:

Question: Check for special characters in text-based columns (e.g., Item Name, Choice Description). How can these be addressed for consistency?


In [28]:
column_check= ['item_name', 'choice_description_1','choice_description_2','choice_description_3','choice_description_4','choice_description_5','choice_description_6','choice_description_7','choice_description_8','choice_description_9','choice_description_10']

In [29]:
# Regular expression to identify special characters
special_chars = re.compile(r'[^a-zA-Z0-9\s]')

In [31]:
# Apply handling to specified columns using applymap and lambda
df[column_check] = df[column_check].applymap(lambda x: special_chars.sub('', x) if isinstance(x, str) else x)

In [32]:
df.head(10)

order_id  quantity                             item_name  item_price  \
0         1         1          Chips and Fresh Tomato Salsa        2.39   
1         1         1                                  Izze        3.39   
2         1         1                      Nantucket Nectar        3.39   
3         1         1  Chips and TomatilloGreen Chili Salsa        2.39   
4         2         2                          Chicken Bowl       16.98   
5         3         1                          Chicken Bowl       10.98   
6         3         1                         Side of Chips        1.69   
7         4         1                         Steak Burrito       11.75   
8         4         1                      Steak Soft Tacos        9.25   
9         5         1                         Steak Burrito        9.25   

           choice_description_1 choice_description_2 choice_description_3  \
0                          None                 None                 None   
1                    Clementine                 None                 None   
2                         Apple                 None                 None   
3                          None                 None                 None   
4  TomatilloRed Chili Salsa Hot          Black Beans                 Rice   
5       Fresh Tomato Salsa Mild                 Rice               Cheese   
6                          None                 None                 None   
7     Tomatillo Red Chili Salsa    Fajita Vegetables          Black Beans   
8   Tomatillo Green Chili Salsa          Pinto Beans               Cheese   
9            Fresh Tomato Salsa                 Rice          Black Beans   

  choice_description_4 choice_description_5 choice_description_6  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4               Cheese           Sour Cream                 None   
5           Sour Cream            Guacamole              Lettuce   
6                 None                 None                 None   
7          Pinto Beans               Cheese           Sour Cream   
8           Sour Cream              Lettuce                 None   
9          Pinto Beans               Cheese           Sour Cream   

  choice_description_7 choice_description_8 choice_description_9  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4                 None                 None                 None   
5                 None                 None                 None   
6                 None                 None                 None   
7            Guacamole              Lettuce                 None   
8                 None                 None                 None   
9              Lettuce                 None                 None   

  choice_description_10  
0                  None  
1                  None  
2                  None  
3                  None  
4                  None  
5                  None  
6                  None  
7                  None  
8                  None  
9                  None

In [33]:
# Checking for special characters in specified columns using str.contains()
special_chars_check = df[column_check].apply(lambda col: col.str.contains(special_chars))
special_chars_check

item_name  choice_description_1 choice_description_2  \
0         False                 False                 None   
1         False                 False                 None   
2         False                 False                 None   
3         False                 False                 None   
4         False                 False                False   
...         ...                   ...                  ...   
4617      False                 False                False   
4618      False                 False                False   
4619      False                 False                False   
4620      False                 False                False   
4621      False                 False                False   

     choice_description_3 choice_description_4 choice_description_5  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                   False                False                False   
...                   ...                  ...                  ...   
4617                False                False                False   
4618                False                False                False   
4619                False                False                False   
4620                False                 None                 None   
4621                False                False                 None   

     choice_description_6 choice_description_7 choice_description_8  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    None                 None                 None   
...                   ...                  ...                  ...   
4617                False                False                 None   
4618                False                 None                 None   
4619                 None                 None                 None   
4620                 None                 None                 None   
4621                 None                 None                 None   

     choice_description_9 choice_description_10  
0                    None                  None  
1                    None                  None  
2                    None                  None  
3                    None                  None  
4                    None                  None  
...                   ...                   ...  
4617                 None                  None  
4618                 None                  None  
4619                 None                  None  
4620                 None                  None  
4621                 None                  None  

[4505 rows x 11 columns]

In [34]:
df.head()

order_id  quantity                             item_name  item_price  \
0         1         1          Chips and Fresh Tomato Salsa        2.39   
1         1         1                                  Izze        3.39   
2         1         1                      Nantucket Nectar        3.39   
3         1         1  Chips and TomatilloGreen Chili Salsa        2.39   
4         2         2                          Chicken Bowl       16.98   

           choice_description_1 choice_description_2 choice_description_3  \
0                          None                 None                 None   
1                    Clementine                 None                 None   
2                         Apple                 None                 None   
3                          None                 None                 None   
4  TomatilloRed Chili Salsa Hot          Black Beans                 Rice   

  choice_description_4 choice_description_5 choice_description_6  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4               Cheese           Sour Cream                 None   

  choice_description_7 choice_description_8 choice_description_9  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4                 None                 None                 None   

  choice_description_10  
0                  None  
1                  None  
2                  None  
3                  None  
4                  None

## 7. Order Id Integrity:
Question: Cross-reference the Order ID column for integrity. Are there any irregularities or patterns that need validation?

In [36]:
# Checkiing for unique Order Ids
unique_orders = df['order_id'].unique()


In [38]:
# Checking for irregularities
if len(unique_orders) == df.shape[0]:
    print("The order IDs are unique and consistent.")
else:
    print("There are irregularities or duplicate order IDs.")

There are irregularities or duplicate order IDs.


In [39]:
# Checking for missing values
missing_values = df['order_id'].isnull().sum()
if missing_values > 0:
    print(f"There are {missing_values} missing order IDs.")
else:
    print("There are No missing Order IDs.")

There are No missing Order IDs.


In [40]:
# Checking for patterns or irregularities in Order IDs
order_id_counts = df['order_id'].value_counts()
print("Counts of each Order ID:")
print(order_id_counts)

Counts of each Order ID:
926     20
1483    14
1786    11
691     11
759     11
        ..
1334     1
1336     1
1339     1
59       1
1004     1
Name: order_id, Length: 1827, dtype: int64


In [41]:
#checking for order_Id integrity
df[df['order_id'] == 691]

order_id  quantity           item_name  item_price  \
1707       691         1     Chicken Burrito        8.75   
1708       691         1  Chicken Salad Bowl       11.25   
1709       691         1       Steak Burrito        9.25   
1710       691         1     Chicken Burrito        8.75   
1711       691         1        Chicken Bowl       11.25   
1712       691         1     Chicken Burrito       11.25   
1713       691         1       Steak Burrito       11.75   
1714       691         1     Chicken Burrito       11.25   
1715       691         1       Steak Burrito       11.75   
1716       691         1        Chicken Bowl       11.25   
1717       691         1       Steak Burrito       11.75   

           choice_description_1 choice_description_2 choice_description_3  \
1707  Tomatillo Red Chili Salsa                 Rice               Cheese   
1708   Roasted Chili Corn Salsa    Fajita Vegetables          Black Beans   
1709  Tomatillo Red Chili Salsa                 Rice          Black Beans   
1710  Tomatillo Red Chili Salsa                 Rice          Pinto Beans   
1711         Fresh Tomato Salsa                 Rice          Black Beans   
1712   Roasted Chili Corn Salsa    Fajita Vegetables                 Rice   
1713         Fresh Tomato Salsa                 Rice          Pinto Beans   
1714   Roasted Chili Corn Salsa                 Rice          Pinto Beans   
1715         Fresh Tomato Salsa                 Rice          Pinto Beans   
1716   Roasted Chili Corn Salsa    Fajita Vegetables                 Rice   
1717         Fresh Tomato Salsa                 Rice          Black Beans   

     choice_description_4 choice_description_5 choice_description_6  \
1707           Sour Cream                 None                 None   
1708            Guacamole              Lettuce                 None   
1709               Cheese              Lettuce                 None   
1710               Cheese              Lettuce                 None   
1711               Cheese           Sour Cream            Guacamole   
1712          Black Beans            Guacamole                 None   
1713               Cheese            Guacamole              Lettuce   
1714               Cheese           Sour Cream            Guacamole   
1715            Guacamole              Lettuce                 None   
1716          Pinto Beans               Cheese            Guacamole   
1717               Cheese           Sour Cream            Guacamole   

     choice_description_7 choice_description_8 choice_description_9  \
1707                 None                 None                 None   
1708                 None                 None                 None   
1709                 None                 None                 None   
1710                 None                 None                 None   
1711                 None                 None                 None   
1712                 None                 None                 None   
1713                 None                 None                 None   
1714              Lettuce                 None                 None   
1715                 None                 None                 None   
1716              Lettuce                 None                 None   
1717              Lettuce                 None                 None   

     choice_description_10  
1707                  None  
1708                  None  
1709                  None  
1710                  None  
1711                  None  
1712                  None  
1713                  None  
1714                  None  
1715                  None  
1716                  None  
1717                  None

## 8. Item Name Standardization:

Question: Standardize the Item Name column. Are there variations that can be unified for better analysis?

In [42]:
df['item_name']

0               Chips and Fresh Tomato Salsa
1                                       Izze
2                           Nantucket Nectar
3       Chips and TomatilloGreen Chili Salsa
4                               Chicken Bowl
                        ...                 
4617                           Steak Burrito
4618                           Steak Burrito
4619                      Chicken Salad Bowl
4620                      Chicken Salad Bowl
4621                      Chicken Salad Bowl
Name: item_name, Length: 4505, dtype: object

In [43]:
# Display unique values in the 'Item_Name' column
unique_item_names = df['item_name'].unique()
print(unique_item_names)


['Chips and Fresh Tomato Salsa' 'Izze' 'Nantucket Nectar'
 'Chips and TomatilloGreen Chili Salsa' 'Chicken Bowl' 'Side of Chips'
 'Steak Burrito' 'Steak Soft Tacos' 'Chips and Guacamole'
 'Chicken Crispy Tacos' 'Chicken Soft Tacos' 'Chicken Burrito'
 'Canned Soda' 'Barbacoa Burrito' 'Carnitas Burrito' 'Carnitas Bowl'
 'Bottled Water' 'Chips and Tomatillo Green Chili Salsa' 'Barbacoa Bowl'
 'Chips' 'Chicken Salad Bowl' 'Steak Bowl' 'Barbacoa Soft Tacos'
 'Veggie Burrito' 'Veggie Bowl' 'Steak Crispy Tacos'
 'Chips and Tomatillo Red Chili Salsa' 'Barbacoa Crispy Tacos'
 'Veggie Salad Bowl' 'Chips and Roasted ChiliCorn Salsa'
 'Chips and Roasted Chili Corn Salsa' 'Carnitas Soft Tacos'
 'Chicken Salad' 'Canned Soft Drink' 'Steak Salad Bowl'
 '6 Pack Soft Drink' 'Chips and TomatilloRed Chili Salsa' 'Bowl' 'Burrito'
 'Crispy Tacos' 'Carnitas Crispy Tacos' 'Steak Salad'
 'Chips and Mild Fresh Tomato Salsa' 'Veggie Soft Tacos'
 'Carnitas Salad Bowl' 'Barbacoa Salad Bowl' 'Salad' 'Veggie Crispy 

In [44]:
# Defining a regex pattern to match non-alphanumeric characters
pattern = r'[^a-zA-Z0-9\s]'

In [45]:
# Displaying cleaned 'Item_Name' column
df['item_name']

0               Chips and Fresh Tomato Salsa
1                                       Izze
2                           Nantucket Nectar
3       Chips and TomatilloGreen Chili Salsa
4                               Chicken Bowl
                        ...                 
4617                           Steak Burrito
4618                           Steak Burrito
4619                      Chicken Salad Bowl
4620                      Chicken Salad Bowl
4621                      Chicken Salad Bowl
Name: item_name, Length: 4505, dtype: object

In [46]:
# Removing special characters from 'Item_Name' column directly
df['item_name']=df['item_name'].str.replace(pattern, '', regex=True)
df['item_name']

0               Chips and Fresh Tomato Salsa
1                                       Izze
2                           Nantucket Nectar
3       Chips and TomatilloGreen Chili Salsa
4                               Chicken Bowl
                        ...                 
4617                           Steak Burrito
4618                           Steak Burrito
4619                      Chicken Salad Bowl
4620                      Chicken Salad Bowl
4621                      Chicken Salad Bowl
Name: item_name, Length: 4505, dtype: object

## **9.** Quantity and Price Relationships And **13.** Consistent Quantity and Price Units:

Question: Investigate the relationships between Quantity and Item Price. Are there cases where adjustments need to be made for accurate analysis?

In [47]:
df[['quantity','item_price']]

quantity  item_price
0            1        2.39
1            1        3.39
2            1        3.39
3            1        2.39
4            2       16.98
...        ...         ...
4617         1       11.75
4618         1       11.75
4619         1       11.25
4620         1        8.75
4621         1        8.75

[4505 rows x 2 columns]

In [48]:
# Summary statistics for Quantity and Item_Price columns
df[['quantity', 'item_price']].describe()

quantity   item_price
count  4505.000000  4505.000000
mean      1.076138     7.514337
std       0.405580     4.243410
min       1.000000     1.090000
25%       1.000000     3.990000
50%       1.000000     8.750000
75%       1.000000     9.390000
max      15.000000    44.250000

In [50]:
# Finding unique items and their costs
item_costs = df.groupby('item_name')['item_price'].mean().to_dict()
item_costs

{'6 Pack Soft Drink': 6.610185185185185,
 'Barbacoa Bowl': 10.2165625,
 'Barbacoa Burrito': 9.845505617977528,
 'Barbacoa Crispy Tacos': 10.928181818181818,
 'Barbacoa Salad Bowl': 10.9525,
 'Barbacoa Soft Tacos': 10.0184,
 'Bottled Water': 1.836027397260274,
 'Bowl': 14.8,
 'Burrito': 7.400000000000001,
 'Canned Soda': 1.3298,
 'Canned Soft Drink': 1.4724199288256228,
 'Carnitas Bowl': 10.833970588235294,
 'Carnitas Burrito': 10.13271186440678,
 'Carnitas Crispy Tacos': 11.137142857142859,
 'Carnitas Salad': 8.99,
 'Carnitas Salad Bowl': 11.056666666666667,
 'Carnitas Soft Tacos': 9.3985,
 'Chicken Bowl': 10.13472496473907,
 'Chicken Burrito': 10.098923933209647,
 'Chicken Crispy Tacos': 10.04531914893617,
 'Chicken Salad': 9.01,
 'Chicken Salad Bowl': 11.170454545454545,
 'Chicken Soft Tacos': 9.659906542056074,
 'Chips': 2.3484878048780486,
 'Chips and Fresh Tomato Salsa': 3.285090909090909,
 'Chips and Guacamole': 4.5981663113006395,
 'Chips and Mild Fresh Tomato Salsa': 3.0,
 'Chi

In [53]:
# Calculating expected price based on Quantity and mean cost per item
expected_price = df['quantity'] * df['item_name'].map(item_costs)

# Checking and replacing inconsistent prices
inconsistent_prices = abs(expected_price - df['item_price']) > 0.01  # Setting a threshold for inconsistency
df.loc[inconsistent_prices, 'item_price'] = expected_price[inconsistent_prices]

df['item_price']=df['item_price'].round(2)

In [54]:
#Rounding off the decimals of Item Price Column
df['item_price']=df['item_price'].round(2)
df

order_id  quantity                             item_name  item_price  \
0            1         1          Chips and Fresh Tomato Salsa        3.29   
1            1         1                                  Izze        3.39   
2            1         1                      Nantucket Nectar        3.64   
3            1         1  Chips and TomatilloGreen Chili Salsa        2.54   
4            2         2                          Chicken Bowl       20.27   
...        ...       ...                                   ...         ...   
4617      1833         1                         Steak Burrito       10.46   
4618      1833         1                         Steak Burrito       10.46   
4619      1834         1                    Chicken Salad Bowl       11.17   
4620      1834         1                    Chicken Salad Bowl       11.17   
4621      1834         1                    Chicken Salad Bowl       11.17   

              choice_description_1 choice_description_2 choice_description_3  \
0                             None                 None                 None   
1                       Clementine                 None                 None   
2                            Apple                 None                 None   
3                             None                 None                 None   
4     TomatilloRed Chili Salsa Hot          Black Beans                 Rice   
...                            ...                  ...                  ...   
4617            Fresh Tomato Salsa                 Rice          Black Beans   
4618            Fresh Tomato Salsa                 Rice           Sour Cream   
4619            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   
4620            Fresh Tomato Salsa    Fajita Vegetables              Lettuce   
4621            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   

     choice_description_4 choice_description_5 choice_description_6  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                  Cheese           Sour Cream                 None   
...                   ...                  ...                  ...   
4617           Sour Cream               Cheese              Lettuce   
4618               Cheese              Lettuce            Guacamole   
4619            Guacamole              Lettuce                 None   
4620                 None                 None                 None   
4621              Lettuce                 None                 None   

     choice_description_7 choice_description_8 choice_description_9  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    None                 None                 None   
...                   ...                  ...                  ...   
4617            Guacamole                 None                 None   
4618                 None                 None                 None   
4619                 None                 None                 None   
4620                 None                 None                 None   
4621                 None                 None                 None   

     choice_description_10  
0                     None  
1                     None  
2                     None  
3                     None  
4                     None  
...                    ...  
4617                  None  
4618                  None  
4619                  None  
4620                  None  
4621                  None  

[4505 rows x 14 columns]

## 12. Handling Categorical Data:
Question: For categorical columns (e.g., Item Name), consider encoding or transforming them into a format suitable for analysis.

In [55]:
df['item_name'].head(20)

0             Chips and Fresh Tomato Salsa
1                                     Izze
2                         Nantucket Nectar
3     Chips and TomatilloGreen Chili Salsa
4                             Chicken Bowl
5                             Chicken Bowl
6                            Side of Chips
7                            Steak Burrito
8                         Steak Soft Tacos
9                            Steak Burrito
10                     Chips and Guacamole
11                    Chicken Crispy Tacos
12                      Chicken Soft Tacos
13                            Chicken Bowl
14                     Chips and Guacamole
15    Chips and TomatilloGreen Chili Salsa
16                         Chicken Burrito
17                         Chicken Burrito
18                             Canned Soda
19                            Chicken Bowl
Name: item_name, dtype: object

In [56]:
#Importing the LabelEncodder
from sklearn.preprocessing import LabelEncoder

In [57]:
# Initializing the LabelEncoder
label_encoder = LabelEncoder()

In [58]:
df.head()

order_id  quantity                             item_name  item_price  \
0         1         1          Chips and Fresh Tomato Salsa        3.29   
1         1         1                                  Izze        3.39   
2         1         1                      Nantucket Nectar        3.64   
3         1         1  Chips and TomatilloGreen Chili Salsa        2.54   
4         2         2                          Chicken Bowl       20.27   

           choice_description_1 choice_description_2 choice_description_3  \
0                          None                 None                 None   
1                    Clementine                 None                 None   
2                         Apple                 None                 None   
3                          None                 None                 None   
4  TomatilloRed Chili Salsa Hot          Black Beans                 Rice   

  choice_description_4 choice_description_5 choice_description_6  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4               Cheese           Sour Cream                 None   

  choice_description_7 choice_description_8 choice_description_9  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4                 None                 None                 None   

  choice_description_10  
0                  None  
1                  None  
2                  None  
3                  None  
4                  None

In [59]:
# Encoding the 'Item_Name' column
encoded_items = label_encoder.fit_transform(df['item_name'])

In [61]:
# Displaying the updated DataFrame with the encoded column
df['Encoded_Item'] = encoded_items
(df)

order_id  quantity                             item_name  item_price  \
0            1         1          Chips and Fresh Tomato Salsa        3.29   
1            1         1                                  Izze        3.39   
2            1         1                      Nantucket Nectar        3.64   
3            1         1  Chips and TomatilloGreen Chili Salsa        2.54   
4            2         2                          Chicken Bowl       20.27   
...        ...       ...                                   ...         ...   
4617      1833         1                         Steak Burrito       10.46   
4618      1833         1                         Steak Burrito       10.46   
4619      1834         1                    Chicken Salad Bowl       11.17   
4620      1834         1                    Chicken Salad Bowl       11.17   
4621      1834         1                    Chicken Salad Bowl       11.17   

              choice_description_1 choice_description_2 choice_description_3  \
0                             None                 None                 None   
1                       Clementine                 None                 None   
2                            Apple                 None                 None   
3                             None                 None                 None   
4     TomatilloRed Chili Salsa Hot          Black Beans                 Rice   
...                            ...                  ...                  ...   
4617            Fresh Tomato Salsa                 Rice          Black Beans   
4618            Fresh Tomato Salsa                 Rice           Sour Cream   
4619            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   
4620            Fresh Tomato Salsa    Fajita Vegetables              Lettuce   
4621            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   

     choice_description_4 choice_description_5 choice_description_6  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                  Cheese           Sour Cream                 None   
...                   ...                  ...                  ...   
4617           Sour Cream               Cheese              Lettuce   
4618               Cheese              Lettuce            Guacamole   
4619            Guacamole              Lettuce                 None   
4620                 None                 None                 None   
4621              Lettuce                 None                 None   

     choice_description_7 choice_description_8 choice_description_9  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    None                 None                 None   
...                   ...                  ...                  ...   
4617            Guacamole                 None                 None   
4618                 None                 None                 None   
4619                 None                 None                 None   
4620                 None                 None                 None   
4621                 None                 None                 None   

     choice_description_10  Encoded_Item  
0                     None            24  
1                     None            34  
2                     None            35  
3                     None            31  
4                     None            17  
...                    ...           ...  
4617                  None            39  
4618                  None            39  
4619           

In [63]:
# Mapping encoded integers back to item names
decoded_mapping = {label: item for label, item in zip(encoded_items, df['item_name'])}

In [65]:
# Encoding the 'Item_Name' column
encoded_items = label_encoder.fit_transform(df['item_name'])

In [66]:
# Displaying the mapping of encoded integers back to item names
for label, item in decoded_mapping.items():
    print(f"Encoded: {label} -> item: {item}")

Encoded: 24 -> item: Chips and Fresh Tomato Salsa
Encoded: 34 -> item: Izze
Encoded: 35 -> item: Nantucket Nectar
Encoded: 31 -> item: Chips and TomatilloGreen Chili Salsa
Encoded: 17 -> item: Chicken Bowl
Encoded: 37 -> item: Side of Chips
Encoded: 39 -> item: Steak Burrito
Encoded: 43 -> item: Steak Soft Tacos
Encoded: 25 -> item: Chips and Guacamole
Encoded: 19 -> item: Chicken Crispy Tacos
Encoded: 22 -> item: Chicken Soft Tacos
Encoded: 18 -> item: Chicken Burrito
Encoded: 9 -> item: Canned Soda
Encoded: 2 -> item: Barbacoa Burrito
Encoded: 12 -> item: Carnitas Burrito
Encoded: 11 -> item: Carnitas Bowl
Encoded: 6 -> item: Bottled Water
Encoded: 29 -> item: Chips and Tomatillo Green Chili Salsa
Encoded: 1 -> item: Barbacoa Bowl
Encoded: 23 -> item: Chips
Encoded: 21 -> item: Chicken Salad Bowl
Encoded: 38 -> item: Steak Bowl
Encoded: 5 -> item: Barbacoa Soft Tacos
Encoded: 45 -> item: Veggie Burrito
Encoded: 44 -> item: Veggie Bowl
Encoded: 40 -> item: Steak Crispy Tacos
Encoded: 

In [67]:
df.head(30)

order_id  quantity                             item_name  item_price  \
0          1         1          Chips and Fresh Tomato Salsa        3.29   
1          1         1                                  Izze        3.39   
2          1         1                      Nantucket Nectar        3.64   
3          1         1  Chips and TomatilloGreen Chili Salsa        2.54   
4          2         2                          Chicken Bowl       20.27   
5          3         1                          Chicken Bowl       10.13   
6          3         1                         Side of Chips        1.84   
7          4         1                         Steak Burrito       10.46   
8          4         1                      Steak Soft Tacos        9.75   
9          5         1                         Steak Burrito       10.46   
10         5         1                   Chips and Guacamole        4.60   
11         6         1                  Chicken Crispy Tacos       10.05   
12         6         1                    Chicken Soft Tacos        9.66   
13         7         1                          Chicken Bowl       10.13   
14         7         1                   Chips and Guacamole        4.60   
15         8         1  Chips and TomatilloGreen Chili Salsa        2.54   
16         8         1                       Chicken Burrito       10.10   
17         9         1                       Chicken Burrito       10.10   
18         9         2                           Canned Soda        2.66   
19        10         1                          Chicken Bowl       10.13   
20        10         1                   Chips and Guacamole        4.60   
21        11         1                      Barbacoa Burrito        9.85   
22        11         1                      Nantucket Nectar        3.64   
23        12         1                       Chicken Burrito       10.10   
24        12         1                                  Izze        3.39   
25        13         1          Chips and Fresh Tomato Salsa        3.29   
26        13         1                          Chicken Bowl       10.13   
27        14         1                      Carnitas Burrito       10.13   
28        14         1                           Canned Soda        1.33   
29        15         1                       Chicken Burrito       10.10   

                 choice_description_1                choice_description_2  \
0                                None                                None   
1                          Clementine                                None   
2                               Apple                                None   
3                                None                                None   
4        TomatilloRed Chili Salsa Hot                         Black Beans   
5             Fresh Tomato Salsa Mild                                Rice   
6                                None                                None   
7           Tomatillo Red Chili Salsa                   Fajita Vegetables   
8         Tomatillo Green Chili Salsa                         Pinto Beans   
9                  Fresh Tomato Salsa                                Rice   
10                               None                                None   
11           Roasted Chili Corn Salsa                   Fajita Vegetables   
12           Roasted Chili Corn Salsa                                Rice   
13                 Fresh Tomato Salsa                   Fajita Vegetables   
14                               None                                None   
15                               None                                None   
16  TomatilloGreen Chili Salsa Medium                         Pinto Beans   
17            Fresh Tomato Salsa Mild                         Black Beans   
18                             Sprite                                None   
19          Tomatillo Red Chili Salsa                   Fajita Vegetables   
20                            

## 11. Converting to CSV:

Optional Question: If needed, convert the cleaned dataset to a CSV file using the to_csv function with sep=','.

In [68]:
# Saving the cleaned dataset to a CSV file
df.to_csv('chipotle_cleaned_dataset.csv', sep=',', index=False)

In [70]:
cleaned_data=pd.read_csv("chipotle_cleaned_dataset.csv")
cleaned_data

order_id  quantity                             item_name  item_price  \
0            1         1          Chips and Fresh Tomato Salsa        3.29   
1            1         1                                  Izze        3.39   
2            1         1                      Nantucket Nectar        3.64   
3            1         1  Chips and TomatilloGreen Chili Salsa        2.54   
4            2         2                          Chicken Bowl       20.27   
...        ...       ...                                   ...         ...   
4500      1833         1                         Steak Burrito       10.46   
4501      1833         1                         Steak Burrito       10.46   
4502      1834         1                    Chicken Salad Bowl       11.17   
4503      1834         1                    Chicken Salad Bowl       11.17   
4504      1834         1                    Chicken Salad Bowl       11.17   

              choice_description_1 choice_description_2 choice_description_3  \
0                             None                  NaN                  NaN   
1                       Clementine                  NaN                  NaN   
2                            Apple                  NaN                  NaN   
3                             None                  NaN                  NaN   
4     TomatilloRed Chili Salsa Hot          Black Beans                 Rice   
...                            ...                  ...                  ...   
4500            Fresh Tomato Salsa                 Rice          Black Beans   
4501            Fresh Tomato Salsa                 Rice           Sour Cream   
4502            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   
4503            Fresh Tomato Salsa    Fajita Vegetables              Lettuce   
4504            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   

     choice_description_4 choice_description_5 choice_description_6  \
0                     NaN                  NaN                  NaN   
1                     NaN                  NaN                  NaN   
2                     NaN                  NaN                  NaN   
3                     NaN                  NaN                  NaN   
4                  Cheese           Sour Cream                  NaN   
...                   ...                  ...                  ...   
4500           Sour Cream               Cheese              Lettuce   
4501               Cheese              Lettuce            Guacamole   
4502            Guacamole              Lettuce                  NaN   
4503                  NaN                  NaN                  NaN   
4504              Lettuce                  NaN                  NaN   

     choice_description_7 choice_description_8 choice_description_9  \
0                     NaN                  NaN                  NaN   
1                     NaN                  NaN                  NaN   
2                     NaN                  NaN                  NaN   
3                     NaN                  NaN                  NaN   
4                     NaN                  NaN                  NaN   
...                   ...                  ...                  ...   
4500            Guacamole                  NaN                  NaN   
4501                  NaN                  NaN                  NaN   
4502                  NaN                  NaN                  NaN   
4503                  NaN                  NaN                  NaN   
4504                  NaN                  NaN                  NaN   

     choice_description_10  Encoded_Item  
0                      NaN            24  
1                      NaN            34  
2                      NaN            35  
3                      NaN            31  
4                      NaN            17  
...                    ...           ...  
4500                   NaN            39  
4501                   NaN            39  
4502           

## Quantity of each item? (Top 5 itmes available in store)

In [71]:
df.head()
# Grouping by 'item_name', summing quantities, and calculating total price
item_counts = df.groupby('item_name').agg({
    'quantity': 'sum',
    'item_price': 'sum'
}).reset_index()
# Sorting by quantity and price to get top 5 items
top_5_by_quantity = item_counts.sort_values(by='quantity', ascending=False).head(5)
top_5_by_price = item_counts.sort_values(by='item_price', ascending=False).head(5)

print("Top 5 items by Quantity:")
print(top_5_by_quantity)

Top 5 items by Quantity:
              item_name  quantity  item_price
17         Chicken Bowl       744     7537.05
18      Chicken Burrito       577     5827.70
25  Chips and Guacamole       496     2281.58
39        Steak Burrito       380     3974.80
10    Canned Soft Drink       331      486.67


In [72]:
print("\nTop 5 items by Total Price:")
print(top_5_by_price)


Top 5 items by Total Price:
              item_name  quantity  item_price
17         Chicken Bowl       744     7537.05
18      Chicken Burrito       577     5827.70
39        Steak Burrito       380     3974.80
38           Steak Bowl       219     2349.78
25  Chips and Guacamole       496     2281.58


## Which is most ordered item in choice_description?

In [74]:
# Concatenating all 'Choice_Description' columns
choice_columns = [col for col in df.columns if 'choice_description_' in col]
all_choices = df[choice_columns].values.flatten()

# Creating a Series with all choices and counting occurrences
choices_series = pd.Series(all_choices)
most_ordered_item = choices_series.value_counts().idxmax()

print("The most ordered item in 'Choice_Description' is:", most_ordered_item)

The most ordered item in 'Choice_Description' is:  Rice


## Calculating total order count

In [75]:
# Calculating total order count
total_orders = df['order_id'].nunique()

print("Total Order Count:", total_orders)

Total Order Count: 1827


## What is revenue?

In [76]:
# Calculating revenue
df['total_price'] = df['quantity'] * df['item_price']
total_revenue = df['total_price'].sum()

print("Total Revenue:", total_revenue)

Total Revenue: 40975.32


## What is average revenue amount per order?

In [77]:
#Calculating number of orders
num_orders = df['order_id'].nunique()

# Calculating average revenue per order
average_revenue_per_order = total_revenue / num_orders

print("Average Revenue Amount per Order:", average_revenue_per_order)

Average Revenue Amount per Order: 22.427651888341543


## what is revenue from each product?

In [78]:
# Calculating total revenue for each product
df['total_price'] = df['quantity'] * df['item_price']
revenue_per_product = df.groupby('item_name')['total_price'].sum()

print("Revenue from Each Product:")
print(revenue_per_product)

Revenue from Each Product:
item_name
6 Pack Soft Drink                         376.77
Barbacoa Bowl                             654.08
Barbacoa Burrito                          876.65
Barbacoa Crispy Tacos                     153.02
Barbacoa Salad Bowl                        87.60
Barbacoa Soft Tacos                       250.50
Bottled Water                             698.14
Bowl                                      148.00
Burrito                                    44.40
Canned Soda                               228.76
Canned Soft Drink                         680.94
Carnitas Bowl                             855.62
Carnitas Burrito                          628.08
Carnitas Crispy Tacos                     111.38
Carnitas Salad                              8.99
Carnitas Salad Bowl                        66.36
Carnitas Soft Tacos                       376.00
Chicken Bowl                             8287.02
Chicken Burrito                          6756.90
Chicken Crispy Tacos            